electrum_nmc/electrum/names.py

```diff
@@ -132,6 +132,63 @@ def validate_value_length(value: bytes):
     if value_length > value_length_limit:
         raise BitcoinException('value length {} exceeds limit of {}'.format(value_length, value_length_limit))
 
+def validate_onion_address(address: str) -> None:
+    # Verify and remove ".onion" from the address if present
+    if not address.endswith(".onion"):
+        raise ValueError("Invalid onion address: missing .onion suffix")
+
+    # Remove ".onion" from the address
+    address = address[:-6]  # Slice slicing to remove the last 6 characters (length of ".onion")
+
+    try:
+        # Initialize the bytearray to hold the decoded service ID
+        decoded_service_id = bytearray([0] * V3_ONION_SERVICE_ID_RAW_SIZE)
+
+        # Decode the service ID from base32 into the decoded_service_id bytearray
+        decoded_service_id = base64.b32decode(address.encode(), decoded_service_id)
+
+        # Check decoded service ID has the expected length
+        if len(decoded_service_id) != V3_ONION_SERVICE_ID_RAW_SIZE:
+            raise ValueError("Invalid service ID length")
+
+        # Check the version byte
+        version_byte = decoded_service_id[V3_ONION_SERVICE_ID_VERSION_OFFSET]
+        if version_byte > 0x03:
+            raise ValueError(f"Warning: this is a v{version_byte} onion service, which is newer than the v3 onion services that Electrum-NMC knows how to validate. Please verify that this is what you really meant to enter.")
+
+        if version_byte != 0x03:
+            raise ValueError(f"Obsolete v{version_byte} onion service")
+
+        # Extract the public key from the decoded service ID
+        public_key = bytearray(decoded_service_id[:ED25519_PUBLIC_KEY_SIZE])
+
+        # Calculate the truncated checksum
+        truncated_checksum = calc_truncated_checksum(public_key)
+
+        # Check if the truncated checksum matches the corresponding bytes in the decoded service ID
+        if truncated_checksum[0] != decoded_service_id[V3_ONION_SERVICE_ID_CHECKSUM_OFFSET] or \
+                truncated_checksum[1] != decoded_service_id[V3_ONION_SERVICE_ID_CHECKSUM_OFFSET + 1]:
+            raise ValueError("Invalid checksum")
+
+    except Exception as e:
+        raise ValueError(f"Invalid onion address: {e}")
+
+def calc_truncated_checksum(public_key):
```

Comment:
Rename the function to have `onion` in its name, so that it's more obvious what it does from the name.